frontend/src/pages/NewPost/component.tsx

```diff
@@ -0,0 +1,100 @@
+import { getRecentPlayedTracks, searchSong } from '@services/SpotifyAPI'
+import { Input, Space, Avatar, List, Button } from 'antd'
+import { useEffect, useState } from 'react'
+import { type components } from '@data/spotify-types'
+import TextArea from 'antd/es/input/TextArea'
+
+const { Search } = Input
+type trackObject = components['schemas']['TrackObject']
+
+export function NewPost(): JSX.Element {
+  const [songs, setSongs] = useState<trackObject[]>([])
+  const [songId, setSongId] = useState<string>('')
+  const [comment, setComment] = useState<string>('')
+
+  useEffect(() => {
+    getRecentPlayedTracks()
+      .then((songs) => {
+        setSongs(songs != null ? songs : [])
+      })
+      .catch((err) => {
+        console.error(err)
+      })
+  }, [])
+
+  const onSearch = async (value: string): Promise<void> => {
+    await searchSong(value)
+      .then((songs) => {
+        setSongs(songs != null ? songs : [])
+      })
+      .catch((err) => {
+        console.error(err)
+      })
+  }
+
+  const onSongClick = (song: trackObject): void => {
+    setSongs([song])
+    setSongId(song.id ?? '')
```

Comment:
hm, when could the songId be null?
Setting an Id to ' ' and sending an empty id value to the backend seems edgy?